packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts

```diff
@@ -173,66 +187,75 @@ export interface IDatePickerProps extends React.Props<DatePicker> {
+   * Error message to render for TextField if input date string parsing fails.
    */
-  shortMonths: string[];
+  invalidInputErrorMessage?: string;
 
   /**
-   * An array of strings for the full names of days of the week.
-   * The array is 0-based, so days[0] should be the full name of Sunday.
+   * Error message to render for TextField if date boundary (minDate, maxDate) validation fails.
    */
-  days: string[];
+  isOutOfBoundsErrorMessage?: string;
 
   /**
-   * An array of strings for the initials of the days of the week.
-   * The array is 0-based, so days[0] should be the initial of Sunday.
+   * An array of strings for the full names of months.
+   * The array is 0-based, so months[0] should be the full name of January.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
-  shortDays: string[];
+  months?: string[];
 
   /**
-   * String to render for button to direct the user to today's date.
+   * An array of strings for the short names of months.
+   * The array is 0-based, so shortMonths[0] should be the short name of January.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
-  goToToday: string;
+  shortMonths?: string[];
 
   /**
-   * Error message to render for TextField if isRequired validation fails.
+   * An array of strings for the full names of days of the week.
+   * The array is 0-based, so days[0] should be the full name of Sunday.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
-  isRequiredErrorMessage?: string;
+  days?: string[];
 
   /**
-   * Error message to render for TextField if input date string parsing fails.
+   * An array of strings for the initials of the days of the week.
+   * The array is 0-based, so days[0] should be the initial of Sunday.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
-  invalidInputErrorMessage?: string;
+  shortDays?: string[];
 
   /**
-   * Error message to render for TextField if date boundary (minDate, maxDate) validation fails.
+   * String to render for button to direct the user to today's date.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
-  isOutOfBoundsErrorMessage?: string;
+  goToToday?: string;
 
   /**
    * Aria-label for the "previous month" button.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
   prevMonthAriaLabel?: string;
 
   /**
    * Aria-label for the "next month" button.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
   nextMonthAriaLabel?: string;
 
   /**
    * Aria-label for the "previous year" button.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
   prevYearAriaLabel?: string;
 
   /**
    * Aria-label for the "next year" button.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
```

Comment:
See my other comment. You should try to rarely ever use deprecated, as changing the existing interface is generally not good.